src/qtt/algorithms/generic.py

```diff
@@ -92,8 +95,13 @@ def subpixelmax(A, mpos, verbose=0):
 
     cy = val
     ay = (valm + valp) / 2 - cy
-    by = ay + cy - valm
-    shift = -by / (2 * ay)  # Maxima of quadradic
+
+    if np.any(ay==0):
+        # degenerate case
+        raise Exception('position specified by mpos is not a maximum')
```

Comment:
In case of 3 equal values we have a maximum, but not a strict maximum. The calculation would fail (division by `ay`), so that is why I added the exception. With a little more thought I think that is more elegant so solve this by then just leaving the maximum as is, instead of raising the error.